### PR TITLE
Implemented POST /encounter endpoint

### PR DESCRIPTION
**Pull request by schi314**
Allow encounter creation through `/encounter` endpoints
* Valid encounter with reference to `Person` returns `201 CREATED`
* Valid encounter without reference to any `Person` returns `400 BAD RESPONSE`
* Invalid encounter returns `400 BAD RESPONSE`
* If `User` is not found, returns `403 FORBIDDEN`

For #41 

**Testing Notes:**
Implemented integration tests and unit tests. 
* Most of test cases passed, with some failed due to some test cases relies on other methods and endpoints (not implemented yet) to properly test

Done manual testing on Postman and endpoints work as expected

### CODE REVIEW

File: backend/src/firebase-configs/forgettable_service_account.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (sher812) [inline backend/src/firebase-configs/forgettable_service_account.json:1]: Might want to remove this file but otherwise I think it all looks good!

Comment (AdamAWiener) [reply backend/src/firebase-configs/forgettable_service_account.json:1]: Agreed, the service account shouldn't be publicly available.

**Comment by AdamAWiener** [comment]
Why is there front-end stuff being added with this merge?

File: backend/src/firebase-configs/forgettable_service_account.json (hunk 1 above)
Comment (schi314) [reply backend/src/firebase-configs/forgettable_service_account.json:1]: True, I'll remove this and the front-end files

**Comment by schi314** [comment]
> Why is there front-end stuff being added with this merge?



> Why is there front-end stuff being added with this merge?

I'm not sure why Git just added those files into my commits. What I did was pulling changes from upstream (that had the changes for the front-end files in this commit), fixed merge conflicts, and commit. Somehow those files were also included in the merge conflict commit as my changes.

I can't seem to remove the changes since it's already committed quite some time ago. Does anybody knows how to fix this, or could I just leave it as it is since it's not being changed?

**Comment by AdamAWiener** [comment]
> > Why is there front-end stuff being added with this merge?
> 
> > Why is there front-end stuff being added with this merge?
> 
> I'm not sure why Git just added those files into my commits. What I did was pulling changes from upstream (that had the changes for the front-end files in this commit), fixed merge conflicts, and commit. Somehow those files were also included in the merge conflict commit as my changes.
> 
> I can't seem to remove the changes since it's already committed quite some time ago. Does anybody knows how to fix this, or could I just leave it as it is since it's not being changed?

Is it possible to only commit the files you changed onto a new branch forked from main and then perform a PR from that? That might be a possible solution.

**Comment by schi314** [comment]
Sounds like a way. I'll close this and open up a new pull request

**Pull request closed without merging**